x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/data/AggregateMetricDoubleBlock.java

```diff
@@ -0,0 +1,292 @@
+            }
+        }
+    }
+
+    @Override
+    public Block keepMask(BooleanVector mask) {
+        AggregateMetricDoubleBlock result = null;
+        DoubleBlock newMinBlock = null;
+        DoubleBlock newMaxBlock = null;
+        DoubleBlock newSumBlock = null;
+        IntBlock newCountBlock = null;
+        try {
+            newMinBlock = minBlock.keepMask(mask);
+            newMaxBlock = maxBlock.keepMask(mask);
+            newSumBlock = sumBlock.keepMask(mask);
+            newCountBlock = countBlock.keepMask(mask);
+            result = new AggregateMetricDoubleBlock(newMinBlock, newMaxBlock, newSumBlock, newCountBlock);
+            return result;
+        } finally {
+            if (result == null) {
+                Releasables.close(newMinBlock, newMaxBlock, newSumBlock, newCountBlock);
+            }
+        }
+    }
+
+    @Override
+    public ReleasableIterator<? extends Block> lookup(IntBlock positions, ByteSizeValue targetBlockSize) {
+        // TODO: support
+        throw new UnsupportedOperationException("can't lookup values from AggregateMetricDoubleBlock");
+    }
+
+    @Override
+    public MvOrdering mvOrdering() {
+        return MvOrdering.UNORDERED;
+    }
+
+    @Override
+    public Block expand() {
+        throw new UnsupportedOperationException("AggregateMetricDoubleBlock");
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        for (Block block : List.of(minBlock, maxBlock, sumBlock, countBlock)) {
+            Block.writeTypedBlock(block, out);
+        }
+    }
+
+    @Override
+    public long ramBytesUsed() {
+        return minBlock.ramBytesUsed() + maxBlock.ramBytesUsed() + sumBlock.ramBytesUsed() + countBlock.ramBytesUsed();
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        AggregateMetricDoubleBlock that = (AggregateMetricDoubleBlock) o;
+        return positionCount == that.positionCount
+            && minBlock.equals(that.minBlock)
+            && maxBlock.equals(that.maxBlock)
+            && sumBlock.equals(that.sumBlock)
+            && countBlock.equals(that.countBlock);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(
+            DoubleBlock.hash(minBlock),
+            DoubleBlock.hash(maxBlock),
+            DoubleBlock.hash(sumBlock),
+            IntBlock.hash(countBlock),
+            positionCount
+        );
+    }
+
+    public static Block readFrom(StreamInput in) throws IOException {
+        boolean success = false;
+        DoubleBlock minBlock = null;
+        DoubleBlock maxBlock = null;
+        DoubleBlock sumBlock = null;
+        IntBlock countBlock = null;
+        BlockStreamInput blockStreamInput = (BlockStreamInput) in;
+        try {
+            minBlock = (DoubleBlock) Block.readTypedBlock(blockStreamInput);
```

Comment:
Can we move the readFrom close to writeTo. Can we also read/write without `typed` since we already know its type. `DoubleBlock.readFrom()`?